sdk/communication/Azure.Communication.Common/src/MicrosoftBotIdentifier.cs

```diff
@@ -0,0 +1,98 @@
+
+        /// <summary>
+        /// Returns the canonical string representation of the <see cref="MicrosoftBotIdentifier"/>.
+        /// You can use the <see cref="RawId"/> for encoding the identifier and then use it as a key in a database.
+        /// </summary>
+        public override string RawId
+        {
+            get
+            {
+                if (_rawId != null)
+                    return _rawId;
+
+                if (IsGlobal)
+                {
+                    if (Cloud == CommunicationCloudEnvironment.Dod)
+                    {
+                        _rawId = $"{BotDodCloudGlobal}{BotId}";
+                    }
+                    else if (Cloud == CommunicationCloudEnvironment.Gcch)
+                    {
+                        _rawId = $"{BotGcchCloudGlobal}{BotId}";
+                    }
+                    else
+                    {
+                        _rawId = $"{Bot}{BotId}";
+                    }
+                }
+                else
+                {
+                    if (Cloud == CommunicationCloudEnvironment.Dod)
+                    {
+                        _rawId = $"{BotDodCloud}{BotId}";
+                    }
+                    else if (Cloud == CommunicationCloudEnvironment.Gcch)
+                    {
+                        _rawId = $"{BotGcchCloud}{BotId}";
+                    }
+                    else
+                    {
+                        _rawId = $"{BotPublicCloud}{BotId}";
+                    }
+                }
+
+                return _rawId;
+            }
+        }
+
+        /// <summary>The id of the Microsoft bot.</summary>
+        public string BotId { get; }
+
+        /// <summary>True if the bot is global and false (or missing) if the bot is tenantized.</summary>
+        public bool IsGlobal { get; }
+
+        /// <summary> The cloud that the bot belongs to. </summary>
+        public CommunicationCloudEnvironment Cloud { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MicrosoftBotIdentifier"/>.
+        /// </summary>
+        /// <param name="botId">Id of the Microsoft bot. The unique Microsoft app ID for the bot as registered with the Bot Framework.</param>
+        /// <param name="isGlobal">Set this to true if the bot is global.</param>
+        /// <param name="cloud">The cloud that the Microsoft bot belongs to. A null value translates to the Public cloud.</param>
+        /// <param name="rawId">Raw id of the Microsoft bot optional.</param>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when the <paramref name="botId"/> is null.
+        /// </exception>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when the <paramref name="botId"/> is empty.
+        /// </exception>
+        public MicrosoftBotIdentifier(string botId, bool isGlobal = false, CommunicationCloudEnvironment? cloud = null, string rawId = null)
```

Comment:
I don't think we need the `string rawId = null` param here. For `MicrosoftTeamsUserIdentifier` it was some kind of legacy identifier support.

Reply:
@AikoBB fyi

Reply:
so it means the raw id should be a mandatory parameter for initialization? should we also check for its validness(not null and empty)?

Reply:
no...the rawId shouldn't be there at all. you should use `CommunicationIdentifier.FromRawId()` if you want to initialize the identifier from rawId. we keep the rawId param in the constructors of MSTeamsUserIdentifier and PhoneNumber identifier for backward compatability reasons but we don't need to introduce it for new identifiers.

Reply:
OK, I'll remove this parameter